api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -123,6 +123,10 @@ message DownstreamTlsContext {
   // an accompanying OCSP response or if the response expires at runtime.
   // Defaults to LENIENT_STAPLING
   OcspStaplePolicy ocsp_staple_policy = 8 [(validate.rules).enum = {defined_only: true}];
+
+  // Config for full scanning certificates if there is no cert matches SNI provided by client.
+  // Defaults to false.
```

Comment:
Can you add quite a bit more text here on the implications of this field? I don't think an end user would know what this means and when they would set it or not. You can also ref link to ssl.rst if that is easier and you want to describe it in detail there. I would also document the runtime flag that can be used to change the default.

Reply:
Sure, will do that.

Reply:
I'd like to confirm with you the usage of RUNTIME_GUARD in https://github.com/envoyproxy/envoy/pull/22036#discussion_r1027466175, and then will update.